x-pack/test/fleet_api_integration/apis/settings/update.ts

```diff
@@ -38,6 +38,18 @@ export default function (providerContext: FtrProviderContext) {
     after(async () => {
       await esArchiver.unload('fleet/empty_fleet_server');
     });
+
+    it('should explicetly set port on fleet_server_hosts', async function () {
+      await supertest
+        .put(`/api/fleet/settings`)
+        .set('kbn-xsrf', 'xxxx')
+        .send({ fleet_server_hosts: ['https://test.fr'] })
+        .expect(200);
+
+      const { body: getSettingsRes } = await supertest.get(`/api/fleet/settings`).expect(200);
+      expect(getSettingsRes.item.fleet_server_hosts).to.eql(['https://test.fr:443']);
```

Comment:
For a follow up: What happens in the case of `http://`?

Reply:
it will force the `80` port, here I have tests that should cover most of the possible cases https://github.com/elastic/kibana/blob/master/x-pack/plugins/fleet/server/services/settings.test.ts#L210-L216